source/code/shared/tools/scx_ssl_config/scxsslcert.cpp

```diff
@@ -529,18 +566,24 @@ void SCXSSLCertificate::DoGenerate()
         }
 
         // Cleanup the rest of the resources that may have been allocated internally.
-        OBJ_cleanup();
+        #if OPENSSL_VERSION_NUMBER <= 0x100fffffL // SSL 1.0.x or lower?
+            OBJ_cleanup();
+        #endif
         CONF_modules_unload(1);
-        CRYPTO_cleanup_all_ex_data();
-        ERR_remove_state(0);
+        #if OPENSSL_VERSION_NUMBER <= 0x100fffffL // SSL 1.0.x or lower?
+            CRYPTO_cleanup_all_ex_data();
+            ERR_remove_state(0);
+        #endif
     }
     catch (SCXCoreLib::SCXException & e)
     {
         // Blunt force resource release functions.
-        OBJ_cleanup();
-        CONF_modules_free();
-        CRYPTO_cleanup_all_ex_data();
-        ERR_remove_state(0);
+        #if OPENSSL_VERSION_NUMBER <= 0x100fffffL // SSL 1.0.x or lower?
+            OBJ_cleanup();
+            CONF_modules_free();
+            CRYPTO_cleanup_all_ex_data();
+            ERR_remove_state(0);
+        #endif
```

Comment:
So `scxsslconfig` is a short-lived program. Technically, if there's a memory leak in it, we'd never notice - it's run very infrequently, and runs for a fraction of a second and is then done.

However, for correctness, have you checked for memory leaks? I'm not sure if it's worth it in this case, but it is the correct thing to do.

Think about it. If you wanted to be 100% correct, you should do it. But if you don't feel like it's worth the effort, I can live with that.

My concern here is that the original author was careful to expunge all memory leaks, and then you introduced some (unknowingly). But again, given the constraints of this program, I can live with it if you feel it's not worth it.

Reply:
As openssl 1.1 documentation it is not needed and these interfaces are no more supported.